sale_order_engine_data/reports/sale_order_template.xml

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <template id="report_saleorder_document_engine_data" inherit_id="sale.report_saleorder_document">
+        <xpath expr="//div[@name='expiration_date']" position="after">
+            <div t-if="doc.engine_ref and doc.engine_hours" class="col-auto col-3 mw-100 mb-2" name="engine_data">
+                <span style="display: block">
```

Comment:
Related to general <div> visibility criteria, for both <span> we should individually check their visibility, for example:
```suggestion
                <span t-if="doc.engine_ref" style="display: block">
```
Please change for both two fields